backend/app/testbot/agents/knowledge_retrieval.py

```
@@ -0,0 +1,108 @@
+# agents/knowledge_retrieval.py
+import logging
+from typing import Dict, Any, List, Optional
+from kb import search_kb  # Original search function
+
+logger = logging.getLogger(__name__)
+
+class KnowledgeRetrievalAgent:
+    """Enhanced knowledge retrieval with context awareness"""
+    
+    async def search(self, chatbot_id: str, query: str, context: Dict[str, Any]) -> List[str]:
```

Ruff (0.8.2)

Use `dict` instead of `Dict` for type annotation

Replace with `dict`

(UP006)